shcomplete/repos.py

```diff
@@ -0,0 +1,78 @@
+import time
+import os
+import argparse
+import sys
+
+from github import Github as GitHub, RateLimitExceededException
+
+class GithubShell(object):
+    """
+    Collects the list of repositories with either bash/zsh/fish history files
+    """
+
+    def __init__(self, login_or_token, password=None, **kwargs):
+        kwargs.setdefault("per_page", 30)
+        self._api = GitHub(login_or_token, password=password, **kwargs, timeout=40)
+
+    def get_repos_list(self):
+        """
+        Get the list of repositories
+        """
+
+        repos_fish = []
+        for num, repo in enumerate(self._api.search_code("filename:fish_history size:>0")):
+            repos_fish.append(repo.html_url)
+            print(num + 1, repos_fish[-1])
+            time.sleep(0.5)
+        repos_fish = {x for x in repos_fish if x.endswith("/fish_history")}
+
+        repos_bash = []
+        for num, repo in enumerate(self._api.search_code("filename:bash_history size:>=950 -user:Dahs81")):
+            repos_bash.append(repo.html_url)
+            print(num + 1, repos_bash[-1])
+            time.sleep(0.7)
```

Comment:
Hehe, wow! Why do you need this? At least put this into an instance variable (init in `__init__`), at max remove it.